text/0000-near-shell.md

```diff
@@ -0,0 +1,191 @@
+
+## NEAR Shell Top-Level Commands 
+
+### `near network <command>`
+
+This command category is used to select and configure NEAR networks for a NEAR Shell user.  This category manipulates the user's `~/.near` configuration to allow a user to specify NEAR network details via the CLI instead of manually editing config files.
+
+Sub-commands for `network` might include:
+
+* `near network list` : Display the current list of networks for the current instance of NEAR Shell.  
+* `near network status` :  Show the user's current network configuration.
+* `near network select` : Select a NEAR network as the default network for subsequent NEAR Shell commands as well as the network configuration for a local server validator node managed by this instance of NEAR Shell.
+* `near network add` : Add a network to the users's `~/.near` config files.
+* `near network remove` : Remove a network from tthe `~/.near` config files.
+* `near network monitor` : Interactively monitor one or more NEAR networks in a curses-like interface updated in a specified interval, default of 1s.
+* `near network help` : Display help on network subcommands. 
+
+### `near account <command>`
+
+This category is used to create, select, and configure accounts on NEAR networks for a given NEAR Shell user.  
+
+* `near account list`   : Display list of accounts configured on local host.
+* `near account status` : Display status of active or specified account.
+* `near account create` : Create a new account or sub-account.
+* `near account delete` : Delete an account or sub-account.
+* `near account select` : Select an account from the list of locally configured accounts as the active account.
+* `near account login`  : Log in the current active or specified account.
+* `near account logout` : Log out the current active or specified account.
+* `near account help`   : Display help on account subcommands.
+
+### `server`
+
+These commands are used to administer a NEAR server validator node.  Currently, only one server per NEAR Shell is proposed; however, multiple servers per physical host should be considered.
+
+* `near server status`  : Display status of NEAR validator server on current host.
+* `near server start`   : Start NEAR validator server on current host.
+* `near server stop`    : Stop NEAR validator server on current host.
+* `near server monitor` : Interactively display NEAR validator server status on current host.
+* `near server tail`    : Tail the log of the NEAR validator server on current host.
+* `near server help`    : Display help on server subcommands.
+
+### `near tokens <command>`
+
+* `near tokens send`	: Send tokens to another NEAR account.
+* `near tokens status`	: Display tokens in wallet.
+* `near tokens stake`
+* `near tokens help`
+
+### `near key <command>`
+
+### `near contract <command>` 
+* `near contract list`
```

Comment:
This would be incredibly useful. Would this information be stored from the user's history? Is there some way we can use a library/RPC to get this info?

Reply:
Not sure what this is supposed to do. Will it just return a list of contract account ids, or contract metadata, or contract code itself?

Reply:
What is the unique identifier for a contract?  Do contracts have names?  The idea was to list contracts for a given account.  It would be good to have an ID and an contract name.

Reply:
The unique identifier is the account name, which is a regular, human-readable NEAR account. Thanks for bringing this up, Bowen. I am considering removing that list of commands.

Reply:
@damons what do you mean by "contracts for a given account"? There can be zero or one contract deployed on a given account. If you are talking about contracts that an account interact with, that is much harder to do because user data are stored under the contract account, not the user account.